### PR TITLE
aws - bedrock custom models

### LINKED ISSUE #9133: Support for AWS Bedrock
### Describe the feature

Would love to have support for AWS Bedrock

### Extra information or context

_No response_

### PR DESCRIPTION
**Pull request by PratMis**
Closes #9133

### CODE REVIEW

**Comment by PratMis** [comment]
Looks like there's one test in the CC provider that's failing

**Comment by ajkerrigan** [comment]
Be resilient to upstream API additions
```diff
diff --git a/tools/c7n_awscc/tests/test_ccmanager.py b/tools/c7n_awscc/tests/test_ccmanager.py
index 2b00bd9df5..531443980b 100644
--- a/tools/c7n_awscc/tests/test_ccmanager.py
+++ b/tools/c7n_awscc/tests/test_ccmanager.py
@@ -13,7 +13,7 @@ def test_init_resource_access_analyzer():
 def test_update_schema():
     klass = initialize_resource("eks_cluster")["EksCluster"]
     update_schema = get_update_schema(klass.schema, "eks_cluster")
-    assert set(update_schema["properties"]) == {
+    assert set(update_schema["properties"]) >= {
         "Tags",
         "Version",
         "ResourcesVpcConfig",

```

**Comment by ajkerrigan** [comment]
> I think this is related to #9158 & #9146, and I wonder if instead of (or in addition to) a meta test with some shape inspection, we could make the test more resilient to upstream API additions without sacrificing its value. Like:

#9165 should take care of the failing awscc test.

**Review by ajkerrigan — CHANGES_REQUESTED** [review]
Functionality looks good to me, couple minor suggestions inline. Thanks for working on this :+1:

File: c7n/resources/bedrock.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+from c7n.manager import resources
+from c7n.query import QueryResourceManager, TypeInfo
+from c7n.tags import RemoveTag, Tag, TagActionFilter, TagDelayedAction
+from c7n.utils import local_session, type_schema
+from c7n.actions import BaseAction
+
+
+@resources.register('bedrock-custom-models')
```
[the remark this answers is not in the document]
Comment (PratMis) [reply c7n/resources/bedrock.py]: Yup, this makes sense. Fixed it, thanks for flagging

File: c7n/resources/bedrock.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+class BedrockCustomModels(QueryResourceManager):
+    class resource_type(TypeInfo):
+        service = 'bedrock'
+        enum_spec = ('list_custom_models', 'modelSummaries[]', None)
+        detail_spec = (
+            'get_custom_model', 'modelIdentifier', 'modelArn', None)
+        id = name = "modelName"
+        arn = "modelArn"
```
[the remark this answers is not in the document]
Comment (PratMis) [reply c7n/resources/bedrock.py]: Addressed

**Review by ajkerrigan — APPROVED** [review]
Looks good to me, thanks again! :+1:

**Pull request merged**